apps/wpcom-block-editor/src/calypso/features/iframe-bridge-server.js

```diff
@@ -1020,40 +1020,28 @@ function handleInlineHelpButton( calypsoPort ) {
  * @param {MessagePort} calypsoPort Port used for communication with parent frame.
  */
 function handleSiteEditorBackButton( calypsoPort ) {
+	// We use the traversal helper because the target element may be the SVG or an SVG element inside.
+	function traverseToFindLink( element, link, depth = 2 ) {
+		let foundLink = false;
+		while ( depth >= 0 ) {
+			if ( element.tagName.toLowerCase() === 'a' && element?.href === link ) {
+				foundLink = true;
+				break;
+			}
+			element = element.parentElement;
+			depth--;
+		}
+		return foundLink;
+	}
+
 	// have to do this event delegation style because the Editor isn't fully initialized yet.
 	document.getElementById( 'wpwrap' ).addEventListener( 'click', ( event ) => {
-		const clickedElement = event.target;
-
-		const isOldDashboardButton =
```

Comment:
I know we've had the references to OldDashBoard button in this file for a while. I did wonder if this was still needed as a backwards compatibility measure for some reason. Sounds like we do not think this is needed any more.

Reply:
The recursion saves me from having to manually check the same thing over and over again via `parentElement` as [below](#diff-3cd9fa865f1d9d2531b85d0a08341a655b497a3304dfbb78ff787e34ff80556bL1052-L1053)